### DISCUSSION TITLE
Sharing Metadata (Badges)

### DISCUSSION THREAD

**Discussion by Prajwalprakash3722** [Kareadita/Kavita#2789] [Ideas]
I propose adding a feature for external metadata integration.

This feature would allow users share their read status and ratings of content, displayed in the form of a badge similar to GitHub badges.

Features: 

•	Display of read status and ratings as badges on content.
•	Users can embed the link in markdown and utilize the feature effortlessly.

**Comment by majora2007** [comment]
I'm not sure I understand how this relates to Kavita? Kavita already supports scrobbling read status and pulls external metadata like reviews and recommendations. 

You're going to need to be more specific.

**Reply by DieselTech** [reply]
I think they are talking about "badges" like you see here:

![image](https://github.com/Kareadita/Kavita/assets/30128380/f43ece5a-2632-4d7e-81f5-e1e2c631eb5c)


But they want a badge generator for kavita that would allow them to hotlink the badge and it display things related to their reading progress. Could be as simple as showing some stats like "Books Read: 42". It's really just an API endpoint that allows unauthorized connections and returns an image and value. 

Uptime Kuma has a pretty good implementation of this that lets you create badges based on data it's aware of:

![image](https://github.com/Kareadita/Kavita/assets/30128380/aff5dc85-1a57-455d-9d6e-7871ab352908)

**Reply by Prajwalprakash3722** [reply]
yes exactly, the endpoint can just return a svg
